src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java

```diff
@@ -49,9 +64,11 @@ public MesosApi(String masterUrl, String user, String frameworkName)
             .withValue("master-url", ConfigValueFactory.fromAnyRef(masterUrl));
     this.clientSettings = MesosClientSettings.fromConfig(conf);
     system = ActorSystem.create("mesos-scheduler");
+    context = system.dispatcher();
     materializer = ActorMaterializer.create(system);
-
     client = connectClient().get();
+
+    adapter = new SchedulerAdapter(Scheduler.fromClient(client), materializer, context);
```

Comment:
```suggestion
    adapter = new SchedulerAdapter(Scheduler.fromClient(client), materializer, context)
        .asSourceAndSink(SpecsSnapshot.empty);
```

Reply:
did we want to use the asSourceAndSink or asAkkaQueue?

Reply:
I'm not sure. Whatever works best for us. It seems to me that we might want a different interface. Once I'm back I'll talk to @meln1k about it.